Editorial review: Document CSS corner-shape (#40663)
mdn/content: files/en-us/web/css/corner-shape-value/index.md

    - `notch`
      - : Defines a concave (90-degree) square corner. The `notch` keyword is equivalent to `superellipse(-infinity)`.
    - `round`
      - : Defines a convex quarter-ellipse, which is the standard rounded corner created by {{cssxref("border-radius")}} without a `corner-shape` applied. This is the default (initial) value for all `corner-shape` properties. The `round` keyword is equivalent to `superellipse(1)`.
    - `scoop`
      - : Defines a concave quarter-ellipse. The `scoop` keyword is equivalent to `superellipse(-1)`.
    - `square`
      - : Defines a 90-degree convex square corner, which is the default corner shape when no `border-radius` (or `border-radius: 0`) is applied. The `square` keyword is equivalent to `superellipse(infinity)`.
    - `squircle`
      - : Defines a quarter of a "squircle", which is a convex curve in between `round` and `square`. The `squircle` keyword is equivalent to `superellipse(2)`.

> [!NOTE]
> You can smoothly animate between different keyword corner shapes as the animation interpolates between their `superellipse()` equivalents.

## Formal syntax

{{csssyntax}}

## Examples

### `<corner-shape-value>` value comparison

> In this example, we provide a drop-down menu allowing you to select different `corner-shape-value` values and a slider that updates the container's {{cssxref("border-radius")}}. This enables visualizing the effect of the different keywords and `superellipse()` parameter values. We've hidden the code for the example for brevity, but you can find a [full explanation of `corner-shape` values](/en-US/docs/Web/CSS/corner-shape#comparing_corner-shape_values) along with other related examples on the {{cssxref("corner-shape")}} reference page.

Reviewer: ```suggestion
In this example, we provide a drop-down menu allowing you to select different `<corner-shape-value>` values and a slider that updates the container's {{cssxref("border-radius")}}. This enables visualizing the effect of the different keywords and `superellipse()` parameter values. The `corner-shape` property defines the shape of the box’s corners while the region to which the shape is applied is specified by the `border-radius` property. The code is hidden for brevity, but you can find a [full explanation of `corner-shape` values](/en-US/docs/Web/CSS/corner-shape#comparing_corner-shape_values) along with other related examples on the {{cssxref("corner-shape")}} reference page.
```

---
title: <corner-shape-value>
slug: Web/CSS/corner-shape-value
page-type: css-type
browser-compat: css.types.corner-shape-value
sidebar: cssref
---

The **`<corner-shape-value>`** [CSS](/en-US/docs/Web/CSS) [data type](/en-US/docs/Web/CSS/CSS_Values_and_Units/CSS_data_types) describes the shape of a container corner. It is used by the shorthand {{cssxref("corner-shape")}} property and its [constituent properties](/en-US/docs/Web/CSS/corner-shape#constituent_properties) to specify the shape to apply to affected container corners.

## Syntax

The `<corner-shape-value>` data type can take a keyword value that describes a common `superellipse()` corner shape, or a {{cssxref("superellipse()")}} function that defines a custom shape.

### Values

- Keywords for common `superellipse()` values

>   - : The available keyword values are as follows:

Author: I've implemented this change, but did it locally, as I did a bit of fiddling with it.

- Keywords
  - : The available keyword values are as follows:
    - `bevel`
      - : Defines a straight, diagonal corner, which is neither convex nor concave. The `bevel` keyword is equivalent to `superellipse(0)`.
    - `notch`
      - : Defines a 90-degree concave square corner. The `notch` keyword is equivalent to `superellipse(-infinity)`.
    - `round`
      - : Defines a convex ordinary ellipse, which is the standard rounded corner created by {{cssxref("border-radius")}} without a `corner-shape` applied. The `round` keyword is equivalent to `superellipse(1)`. This is the default (initial) value for all `corner-shape` properties.
    - `scoop`
      - : Defines a concave ordinary ellipse. The `scoop` keyword is equivalent to `superellipse(-1)`.
    - `square`
      - : Defines a 90-degree convex square corner, which is the default corner shape when no `border-radius` (or `border-radius: 0`) is applied. The `square` keyword is equivalent to `superellipse(infinity)`.
    - `squircle`
      - : Defines a "squircle", which is a convex curve in between `round` and `square`. The `squircle` keyword is equivalent to `superellipse(2)`.

> [!NOTE]

> > You can smoothly animate between different keyword corner shapes as the animation interpolates between their `superellipse()` equivalents.

Reviewer: what value of K used for square or notch? It couldn't be infinity?

Reviewer: If you animate from 0 to 10, the midpoint is 5. steps(5) would be 8, 6, 4, 2, 0
if you animate from 0 to 10,000,000 the midpoint is 5,000,000. steps(5) would be 8M, 6M, 4M, 2M, and 0

## Formal syntax

{{csssyntax}}

## Examples

### `<corner-shape-value>` value comparison

> In this example, we provide a drop-down menu allowing you to select different `<corner-shape-value>` values and a slider that updates the container's {{cssxref("border-radius")}}. This enables visualizing the effect of the different keywords and `superellipse()` parameter values. The `corner-shape` property defines the shape of the box's corners while the region the shape is applied to is specified by the `border-radius` property. The code is hidden for brevity, but you can find a [full explanation of `corner-shape` values](/en-US/docs/Web/CSS/corner-shape#comparing_corner-shape_values) along with other related examples on the {{cssxref("corner-shape")}} reference page.

Reviewer: ```suggestion
In this example, we provide a drop-down menu allowing you to select different `<corner-shape-value>` values and a slider that updates the container's {{cssxref("border-radius")}} to enable visualizing the effect of the different keywords and `superellipse()` parameter values. The `corner-shape` property defines the shape of the box's corners while the region the shape is applied to is specified by the `border-radius` property. The code is hidden for brevity, but you can find a [full explanation of `corner-shape` values](/en-US/docs/Web/CSS/corner-shape#comparing_corner-shape_values) along with other related examples on the {{cssxref("corner-shape")}} reference page.
```
the paragraph just seemed a bit heavy. Originally I suggested a paragraph break before "The `corner-shape` property defines the ..." to help that important bit stick out, but think this suggestion, merging the first two sentences, helps surface that important sentence while reducing the load of the paragraph.

Author: I'd rather do the paragraph break, which is what I have done in my next commit. Your suggestion here makes that sentence a bit long for comfort.

html {
  font-family: Arial, Helvetica, sans-serif;
}

body {
  width: fit-content;
  margin: 20px auto;
}

section {
  display: flex;
  justify-content: center;
  align-items: center;
  margin-top: 20px;
}

select {
  padding: 3px 5px;
}

form div:nth-of-type(2) {
  margin-top: 5px;
  display: flex;
}

section {
  width: 100%;
  height: 180px;
  background-color: orange;
  background-image: linear-gradient(
    to bottom,
    rgb(255 255 255 / 0),
    rgb(255 255 255 / 0.5)
  );
}

section {
  box-shadow: 1px 1px 3px gray;
}
```

```js hidden live-sample___value-comparison
const rectangle = document.querySelector("section");
const select = document.querySelector("select");
const range = document.getElementById("radius-slider");

function setCorners() {
  rectangle.style.cornerShape = select.value;
  const brValue = `${range.value}px`;
  rectangle.style.borderRadius = brValue;
  rectangle.innerHTML = `<div><code>corner-shape: ${select.value};</code><br><code>border-radius: ${brValue};</code></div>`;
}

select.addEventListener("change", setCorners);
range.addEventListener("input", setCorners);
setCorners();
```

{{EmbedLiveSample("value-comparison", "100%", "300")}}

> [!NOTE]

> > See also our [`superellipse()` function value comparison](/en-US/docs/Web/CSS/superellipse#superellipse_value_comparison) example.

Reviewer: ```suggestion
> See also the [`superellipse()` function value comparison](/en-US/docs/Web/CSS/superellipse#superellipse_value_comparison) example.
```

Author: updated in next commit.
